### PR TITLE
4498 - Automation IDs for autocomplete, field filter, dropdown and popupmenu

### PR DESCRIPTION
**Pull request by jmacaluso711**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
Adds automation is attributes for autocomplete, field filter, dropdown and popupmenu

**Related github/jira issue (required)**:
related to #4498 

**Steps necessary to review your pull request (required)**:
- pull and run branch
- go to http://localhost:4000/components/autocomplete/example-index.html and make sure the element gets automation id from the settings
- go to http://localhost:4000/components/field-filter/example-index.html and make sure the field wrapper, label and select get an automation id from the settings
- go to http://localhost:4000/components/dropdown/example-index.html open the dropdown and make sure the dropdown list, label, input, trigger and listbox get an automation id from the settings
- go to http://localhost:4000/components/popupmenu/example-index.html open the dropdown and make sure the popupmenu and options get an automation id from the settings

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: src/components/autocomplete/autocomplete.js [hunk 1]
```diff
@@ -222,6 +222,8 @@ Autocomplete.prototype = {
       role: 'combobox',
       autocomplete: 'off'
     });
+    // Add test automation ids
+    utils.addAttributes(this.element, this, this.settings.attributes);
```
Comment (tmcconechy) [inline src/components/autocomplete/autocomplete.js:226]: We probably need to add id's to the menu items (like i did in pager). But im not sure if that would be part of dropdown/ menu or autocomplete

File: src/components/autocomplete/readme.md [hunk 2]
```diff
@@ -74,4 +74,28 @@ The data returned should be an array with a specific data structure containing `
...
+
+```js
+  attributes: [{ name: 'id', value: 'my-unique-id' }, { name: 'data-automation-id', value: 'my-unique-id' }]
+```
+
+If setting the id/automation id with a function, the id will be a running total of open autocomplete.
+
+For the autocomplete, `attributes` can be set either on the root settings or in each button element. For the button elements you can also use the `id` as before or the new attributes setting to set an id or automation id on the button.
+
+If you set the attributes on the root message, you will get an ID added to the root of the message dialog. Also the message area will get an id with `-message` appended after the id given. And the h1 area will get an id with `-title` appended after the id given. And finally the close button (if used) with get `btn-close` appended.
```
Comment (tmcconechy) [inline src/components/autocomplete/readme.md:99]: This comment is related to message still.

File: src/components/dropdown/dropdown.js [hunk 3]
```diff
@@ -1861,6 +1861,13 @@ Dropdown.prototype = {
       .attr('aria-expanded', 'true')
       .addClass('is-open');
 
+    // Add test automation ids
+    utils.addAttributes(this.list, this, this.settings.attributes);
+    utils.addAttributes(this.list.find('label'), this, this.settings.attributes, 'label');
+    utils.addAttributes(this.list.find('input'), this, this.settings.attributes, 'input');
+    utils.addAttributes(this.list.find('.trigger'), this, this.settings.attributes, 'trigger');
+    utils.addAttributes(this.list.find('ul'), this, this.settings.attributes, 'listbox');
+
```
Comment (tmcconechy) [inline src/components/dropdown/dropdown.js:1870]: Probably need to do the options. I think this means appending something to the a tags (even if id's work directly there the automation-id wont)

File: src/components/dropdown/readme.md [hunk 4]
```diff
@@ -77,7 +77,31 @@ The dropdown list has been coded to act similar to the [aria best practices coll
...
+
+```js
+  attributes: [{ name: 'id', value: 'my-unique-id' }, { name: 'data-automation-id', value: 'my-unique-id' }]
+```
+
+If setting the id/automation id with a function, the id will be a running total of open dropdown.
+
+For the dropdown, `attributes` can be set either on the root settings or in each button element. For the button elements you can also use the `id` as before or the new attributes setting to set an id or automation id on the button.
+
+If you set the attributes on the root message, you will get an ID added to the root of the message dialog. Also the message area will get an id with `-message` appended after the id given. And the h1 area will get an id with `-title` appended after the id given. And finally the close button (if used) with get `btn-close` appended.
```
Comment (tmcconechy) [inline src/components/dropdown/readme.md:102]: Message component

**Review by tmcconechy — COMMENTED** [review]
A Few suggestions. Also can you check off docs/automation.md and the checks on the issue https://github.com/infor-design/enterprise/issues/4498

File: src/components/autocomplete/autocomplete.js [hunk 5]
```diff
@@ -251,6 +253,15 @@ Autocomplete.prototype = {
       this.list = $('<ul id="autocomplete-list" aria-expanded="true"></ul>').appendTo('body');
     }
 
+    // Add test automation ids
+    utils.addAttributes(this.list, this, this.settings.attributes, 'list');
```
Comment (EdwardCoyle) [inline src/components/autocomplete/autocomplete.js:257]: When we implement attributes on the items in the Popupmenu, this is probably going to conflict with that.  Maybe this belongs in there instead? (could resolve the need for the timeout as well)

**Review by EdwardCoyle — CHANGES_REQUESTED** [review]
Looks good to me but just wondering about one thing:

File: src/components/autocomplete/autocomplete.js (hunk 5 above)
Comment (jmacaluso711) [reply src/components/autocomplete/autocomplete.js:257]: I thought that as well, but it seems like it's necessary to add the attributes in the autocomplete specifically. That said, I added the attributes to the `popupmenu` in this PR

**Review by EdwardCoyle — DISMISSED** [review]

**Review by tmcconechy — DISMISSED** [review]

**Review by tmcconechy — APPROVED** [review]

**Review by EdwardCoyle — APPROVED** [review]

**Pull request merged**